File: 2154_Keep_Multiplying_Found_Values_by_Two/2154.cpp
#include <bits/stdc++.h>
using namespace std;

class Solution {
public:
    int findFinalValue(vector<int>& nums, int original) {
        set<int> n(nums.begin(), nums.end());
        while (n.count(original)) {
            original *= 2;
        }
        return original;
    }
};

int main() {
    vector<int> nums = {5, 3, 6, 1, 12};
    int original = 3;
    Solution a;
    cout << a.findFinalValue(nums, original) << "\n";
    return 0;
}

File: 2154_Keep_Multiplying_Found_Values_by_Two/2154.py
class Solution:
    def findFinalValue(self, nums: list[int], original: int) -> int:
        nums = list(set(nums))
        nums.sort()
        while original in nums:
            original *= 2
        return original

a = Solution()
print(a.findFinalValue(nums = [5,3,6,1,12], original = 3))